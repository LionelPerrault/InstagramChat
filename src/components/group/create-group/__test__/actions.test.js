import React from 'react'
import { create } from 'react-test-renderer'
import CreateGroupActions from '../actions'
import { shallow } from 'enzyme'

describe('CreateGroupActions Component', () => {
  const comp = (
    <CreateGroupActions back={jest.fn()} create={jest.fn()} name="dd" />
  )

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should mock back & create actions', () => {
    const wrapper = shallow(comp)

    // create
    wrapper.find('PrimaryButton').simulate('click', {
      preventDefault() {},
    })

    // back
    wrapper.find('SecondaryButton').simulate('click', {
      preventDefault() {},
    })
  })
})
