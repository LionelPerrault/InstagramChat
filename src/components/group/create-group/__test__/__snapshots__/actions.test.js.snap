// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGroupActions Component should match snapshot 1`] = `
Array [
  <a
    className="sec_btn"
    href="#"
    onClick={[MockFunction]}
  >
    Back
  </a>,
  <a
    className="pri_btn c_g_update"
    href="#"
    onClick={[MockFunction]}
  >
    Create group
  </a>,
]
`;
